controlplane/server/otelcolext/processor.go

```diff
@@ -78,23 +90,150 @@ func newLogsProcessor(cfg *Config, zapLogger *zap.Logger, nextConsumer consumer.
 	}
 
 	return &neblic{
-		cfg:          cfg,
-		logger:       zapLogger,
-		nextConsumer: nextConsumer,
-		serverOpts:   serverOpts,
-		ruleBuilder:  builder,
+		cfg:         cfg,
+		logger:      zapLogger,
+		notifyErr:   func(err error) { zapLogger.Error("error digesting sample", zap.Error(err)) },
+		exporter:    NewExporter(nextConsumer),
+		serverOpts:  serverOpts,
+		ruleBuilder: builder,
+		runtimes:    map[samplerIdentifier]*runtime{},
 	}, nil
 }
 
+func (n *neblic) newDigester(resource string, sampler string) *digest.Digester {
+	return digest.NewDigester(digest.Settings{
+		ResourceName:   resource,
+		SamplerName:    sampler,
+		EnabledDigests: []control.DigestType{control.DigestTypeValue},
+		NotifyErr:      n.notifyErr,
+		Exporter:       n.exporter,
+	})
+}
+
 func (n *neblic) Start(ctx context.Context, host component.Host) error {
 	var err error
 	n.s, err = server.New(n.cfg.UID, n.serverOpts...)
 	if err != nil {
 		return err
 	}
-	n.samplerRegistry = n.s.SamplerRegistry()
 
-	return n.s.Start(n.cfg.Endpoint)
+	err = n.s.Start(n.cfg.Endpoint)
+	if err != nil {
+		return err
+	}
+
+	// Populate existing config from the controlplane
+	n.runtimes = map[samplerIdentifier]*runtime{}
+	n.s.RangeSamplersConfig(func(resource, sampler string, config control.SamplerConfig) (carryon bool) {
```

Comment:
There seems to be some duplicated initialization logic common between this bit and the config update goroutine. It would be better to abstract the common pieces or if instead of using `RangeSamplersConfig` on initialization you make sure all configs are sent as events (as mentioned in another comment), you can have all the logic in the same place.